src/test/cs/Infinispan/HotRod/NearCacheFailoverTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Infinispan.HotRod.Config;
+using Infinispan.HotRod.Tests.Util;
+using Infinispan.HotRod.Transport;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace Infinispan.HotRod.Tests
+{
+    class StickyBalancingStragegy : Infinispan.HotRod.Transport.FailOverRequestBalancingStrategy
+    {
+        IList<InetSocketAddress> servers;
+        public InetSocketAddress nextServer(IList<InetSocketAddress> failedServers)
+        {
+            if (servers.Count == 1)
+                return servers[0];
+            // Always choose the 11322 server, the choose 11222 on failure
+            if (failedServers.Count == 0)
+            {
+                return (servers[0].Port == 11322) ? servers[0] : servers[1];
+            }
+            else
+            {
+                return (servers[0].Port == 11322) ? servers[1] : servers[0];
+            }
+
+            throw new System.Exception("No transport");
+        }
+        public void setServers(IList<InetSocketAddress> servers)
+        {
+            this.servers = servers;
+        }
+    }
+
+    public class NearCacheFailoverTest
+    {
+        RemoteCacheManager remoteManager;
+        IMarshaller marshaller;
+        static StickyBalancingStragegy b;
+        HotRodServer server1;
+        HotRodServer server2;
+
+        FailOverRequestBalancingStrategyProducerDelegate d = delegate ()
+        {
+            b = new StickyBalancingStragegy();
+            return b;
+        };
+
+        [TestFixtureSetUp]
+        public void BeforeClass()
+        {
+            BeforeSuite();
+            ConfigurationBuilder conf = new ConfigurationBuilder();
+            conf.AddServer().Host("127.0.0.1").Port(11222);
+            conf.AddServer().Host("127.0.0.1").Port(11322)
+            .ConnectionTimeout(90000).SocketTimeout(6000)
+            .NearCache().Mode(NearCacheMode.INVALIDATED).MaxEntries(10);
+            conf.BalancingStrategyProducer(d);
+            marshaller = new JBasicMarshaller();
+            conf.Marshaller(marshaller);
+            remoteManager = new RemoteCacheManager(conf.Build(), true);
+        }
+
+        public void BeforeSuite()
```

Comment:
I believe this method should be moved to ClusterTestSuite class and annotated with [TestFixtureSetUp]. This way it will be called Before Suite. You don't have to call it manually from BeforeClass method then.

Reply:
OK. Then maybe just moving the BeforeSuite code to the BeforeClass method, right?